scripts/zkai_d128_snark_receipt_timing_setup_gate.py

```
@@ -0,0 +1,715 @@
+    inventory = require_list(payload["mutation_inventory"], "mutation inventory")
+    expect_equal(tuple((item.get("mutation"), item.get("surface")) for item in inventory), EXPECTED_MUTATION_INVENTORY, "mutation inventory")
+    cases = require_list(payload["cases"], "cases")
+    expect_equal(payload["case_count"], len(cases), "case count")
+    expect_equal(len(cases), len(EXPECTED_MUTATION_INVENTORY), "case length")
+    by_name = {case.get("mutation"): case for case in cases}
+    expect_equal(set(by_name), EXPECTED_MUTATION_SET, "case mutation set")
+    for mutation, surface in EXPECTED_MUTATION_INVENTORY:
+        case = by_name[mutation]
+        expect_equal(case.get("surface"), surface, f"surface {mutation}")
+        if case.get("rejected") is not True:
+            raise D128SnarkTimingSetupError(f"mutation accepted: {mutation}", layer="mutation_suite")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)